src/OpenLoco/src/Drawing/SoftwareDrawingContext.cpp

```diff
@@ -1497,43 +1497,147 @@ namespace OpenLoco::Drawing
         }
 
         // 0x004474BA
-        static void drawRectImpl(Gfx::RenderTarget& rt, int16_t left, int16_t top, int16_t right, int16_t bottom, uint32_t colour)
+        static void drawRectImpl(Gfx::RenderTarget& rt, int16_t left, int16_t top, int16_t right, int16_t bottom, uint8_t colour, RectFlags flags)
         {
             registers regs;
             regs.ax = left;
             regs.bx = right;
             regs.cx = top;
             regs.dx = bottom;
-            regs.ebp = colour;
+            regs.ebp = colour | enumValue(flags);
             regs.edi = X86Pointer(&rt);
             call(0x004474BA, regs);
         }
 
-        static void fillRect(Gfx::RenderTarget& rt, int16_t left, int16_t top, int16_t right, int16_t bottom, uint32_t colour)
+        static void drawRectImpl(Gfx::RenderTarget& rt, const Ui::Rect& rect, uint8_t colour, RectFlags flags)
         {
-            drawRectImpl(rt, left, top, right, bottom, colour);
+            drawRectImpl(rt, rect.left(), rect.top(), rect.right(), rect.bottom(), colour, flags);
         }
 
-        static void drawRect(Gfx::RenderTarget& rt, int16_t x, int16_t y, uint16_t dx, uint16_t dy, uint32_t colour)
+        static void fillRect(Gfx::RenderTarget& rt, int16_t left, int16_t top, int16_t right, int16_t bottom, uint8_t colour, RectFlags flags)
+        {
+            drawRectImpl(rt, left, top, right, bottom, colour, flags);
+        }
+
+        static void drawRect(Gfx::RenderTarget& rt, int16_t x, int16_t y, uint16_t dx, uint16_t dy, uint8_t colour, RectFlags flags)
         {
             // This makes the function signature more like a drawing application
-            drawRectImpl(rt, x, y, x + dx - 1, y + dy - 1, colour);
+            drawRectImpl(rt, x, y, x + dx - 1, y + dy - 1, colour, flags);
         }
 
-        static void fillRectInset(Gfx::RenderTarget& rt, int16_t left, int16_t top, int16_t right, int16_t bottom, uint32_t colour, uint8_t flags)
+        static void fillRectInset(Gfx::RenderTarget& rt, int16_t left, int16_t top, int16_t right, int16_t bottom, AdvancedColour colour, RectInsetFlags flags)
         {
-            registers regs;
-            regs.ax = left;
-            regs.bx = right;
-            regs.cx = top;
-            regs.dx = bottom;
-            regs.ebp = colour;
-            regs.edi = X86Pointer(&rt);
-            regs.si = flags;
-            call(0x004C58C7, regs);
+            const auto rect = Ui::Rect::fromLTRB(left, top, right, bottom);
+            const auto baseColour = static_cast<OpenLoco::Colour>(colour);
```

Comment:
Is the `OpenLoco::` qualifier necessary?

Reply:
No i guess not